test/unit/api/ingest/_data.js

```diff
@@ -0,0 +1,193 @@
+          .then(() => {
+            return es.count({ index: 'names' })
+            .then((res) => {
+              expect(res.count).to.be(100);
+            });
+          });
+        });
+      });
+
+      bdd.it('should stream a chunked response', function () {
+        return request.post('/kibana/names/_data')
+        .attach('csv', 'test/unit/data/fake_names.csv')
+        .expect('Transfer-Encoding', 'chunked')
+        .expect(200);
+      });
+
+      bdd.it('should respond with an array of one or more "result objects"', function () {
+        return request.post('/kibana/names/_data')
+          .attach('csv', 'test/unit/data/fake_names_big.csv')
+          .expect(200)
+          .then((dataResponse) => {
+            expect(dataResponse.body.length).to.be(14);
+          });
+      });
+
+      bdd.describe('result objects', function () {
+
+        bdd.it('should include a count of created documents', function () {
+          return request.post('/kibana/names/_data')
+          .attach('csv', 'test/unit/data/fake_names.csv')
+          .expect(200)
+          .then((dataResponse) => {
+            expect(dataResponse.body[0]).to.have.property('created');
+            expect(dataResponse.body[0].created).to.be(100);
+          });
+        });
+
+        bdd.it('should report any indexing errors per document under an "errors.index" key', function () {
+          return request.post('/kibana/names/_data')
+            .attach('csv', 'test/unit/data/fake_names_with_mapping_errors.csv')
+            .expect(200)
+            .then((dataResponse) => {
+              expect(dataResponse.body[0]).to.have.property('created');
+              expect(dataResponse.body[0].created).to.be(98);
+              expect(dataResponse.body[0]).to.have.property('errors');
+              expect(dataResponse.body[0].errors).to.have.property('index');
+              expect(dataResponse.body[0].errors.index.length).to.be(2);
+            });
+        });
+
+        bdd.it('should report any csv parsing errors under an "errors.other" key', function () {
+          return request.post('/kibana/names/_data')
+            .attach('csv', 'test/unit/data/fake_names_with_parse_errors.csv')
+            .expect(200)
+            .then((dataResponse) => {
+              // parse errors immediately abort indexing
+              expect(dataResponse.body[0]).to.have.property('created');
+              expect(dataResponse.body[0].created).to.be(0);
+
+              expect(dataResponse.body[0]).to.have.property('errors');
+              expect(dataResponse.body[0].errors).to.have.property('other');
+              expect(dataResponse.body[0].errors.other.length).to.be(1);
+            });
+        });
+
+      });
+
+      bdd.describe('optional parameters', function () {
+        bdd.it('should accept a custom delimiter query string param for parsing the CSV', function () {
+          return request.post('/kibana/names/_data?delimiter=|')
```

Comment:
Will the API handle the `delimiter` correctly if it is URL-encoded? For example, in this case it would be `delimiter=%7F`. Mind writing a separate unit test for that?